### PR TITLE
Added Dark Theme

### PR DESCRIPTION
**Pull request by WilsontheWolf**

### CODE REVIEW

**Comment by Alexejhero** [comment]
Seems pretty good, I'll make some small modifications later, then merge it.

**Comment by WilsontheWolf** [comment]
For login fix refer to #8

**Comment by Alexejhero** [comment]
Don't forget to change the file paths from `server.js`!

**Comment by WilsontheWolf** [comment]
already ahead of you. 

They are now ejs. I also can't get footer working

**Comment by Alexejhero** [comment]
I'll update the footer template to make it work

File: www/css/themes/main.css [hunk 1]
```diff
@@ -1,7 +1,7 @@
 body {
```
Comment (Alexejhero) [inline www/css/themes/main.css]: You can change the scrollbar color by modifying the `fill` property of the `blocklyScrollbarHandle` class

image

File: www/css/themes/main.css [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-  --bg-color: black;
-  --txt-color: white;
-  --disabled-color: lightgray;
+  --bg-color: white;
+  --txt-color: black;
+  --disabled-color: darkgray;
 }
```
Comment (Alexejhero) [inline www/css/themes/main.css]: The color of `a` links (both visited and unvisited) should be changed so it matches the dark theme better

image

File: www/css/themes/night.css [hunk 3]
```diff
@@ -0,0 +1,5 @@
+body {
```
Comment (Alexejhero) [inline www/css/themes/night.css]: I deleted light theme since it was 99% like the default one.

In my opinion, the gray theme doesn't look too good, so I would personally remove it.

The night and the blackout theme are very similar, so my opinion would be to keep only one of them.

**Review by Alexejhero — CHANGES_REQUESTED** [review]

**Comment by Alexejhero** [comment]
If we narrow it down to two themes, one of which being the default one, we can swap the select field with a checkbox/toggle.

File: www/css/themes/night.css (hunk 3 above)
Comment (WilsontheWolf) [reply www/css/themes/night.css]: I didn't really like the gray either but it was an option. If i had to choose one i would do the night theme as i like it better than the blackout

File: www/css/themes/main.css (hunk 2 above)
Comment (WilsontheWolf) [reply www/css/themes/main.css]: do you want the color to be this?

Comment (Alexejhero) [reply www/css/themes/main.css]: That is the current color of visited links, and in my opinion is too close to the background color.

**Comment by WilsontheWolf** [comment]
Wait when was the main theme merged with the theme element? I don't think I can get default theme like this

**Comment by Alexejhero** [comment]
The main theme was not merged with the theme element. I just renamed `theme.css` to `themes/main.css`. There are still two different files.

**Comment by WilsontheWolf** [comment]
oh im blind

File: www/css/themes/main.css (hunk 1 above)
Comment (WilsontheWolf) [reply www/css/themes/main.css]: ~~false~~ Edit: read below
image
image

Comment (WilsontheWolf) [reply www/css/themes/main.css]: wait i had an error. it works now 🤦‍♂

**Comment by WilsontheWolf** [comment]
Done. Time to push

File: www/css/dark.css [hunk 4]
```diff
@@ -1,7 +1,8 @@
 body {
```
Comment (Alexejhero) [inline www/css/dark.css:1]: The new footer needs to be styled

image

File: www/css/dark.css [hunk 5]
```diff
@@ -1,7 +1,8 @@
...
-  --bg-color: black;
+  --bg-color: #160f19;
   --txt-color: white;
   --disabled-color: lightgray;
+  --a-color: #0af;
 }
```
Comment (Alexejhero) [inline www/css/dark.css:8]: Guilds without a configuration are greyed out in the admin menu.

This does not work on the dark theme, and needs to be updated.

I'll probably do this myself since I can see what I'm working with.

image

image

File: www/css/dark.css [hunk 6]
```diff
@@ -14,25 +15,21 @@ body {
   background-color: var(--bg-color);
 }
```
Comment (Alexejhero) [inline www/css/dark.css:19]: Shadows should be made darker

image

File: www/css/dark.css [hunk 7]
```diff
@@ -14,25 +15,21 @@ body {
...
 
-.blocklyWorkspace{
+.blocklyWorkspace {
   color: var(--txt-color);
   background-color: var(--bg-color);
 }
```
Comment (Alexejhero) [inline www/css/dark.css:24]: The scroll bars are waaay too bright

image

File: www/css/dark.css [hunk 8]
```diff
@@ -14,25 +15,21 @@ body {
...
 
-.blocklyToolboxDiv{
+.blocklyToolboxDiv {
   color: var(--txt-color);
   background-color: var(--bg-color);
 }
```
Comment (Alexejhero) [inline www/css/dark.css:29]: I don't know, it just seems like there's something about the blocks that makes them more... pale.

I know it's an optical illusion, but could you make the blocks look more... alive on the dark theme?

It may just be me, and there might be no actual difference between the blocks on dark theme and the blocks on light theme :joy:

image

**Review by Alexejhero — DISMISSED** [review]
Here are some things that _I think_ need to be changed about the dark theme.

These decisions are not final and are open for discussion.

I might end up doing some of these myself.

File: www/css/dark.css (hunk 5 above)
Comment (WilsontheWolf) [reply www/css/dark.css:8]: why are they hyperlinked if your just going to gray them out?

File: www/css/dark.css (hunk 8 above)
Comment (WilsontheWolf) [reply www/css/dark.css:29]: the blocks are not changed at all

File: www/html/partials/blockly.ejs [hunk 9]
```diff
@@ -1,11 +1,14 @@
-<% include head %>
```
Comment (WilsontheWolf) [inline www/html/partials/blockly.ejs:1]: ~~Why was this removed? This is the theme stuff.~~ nvm the footer has it

File: www/css/dark.css (hunk 5 above)
Comment (Alexejhero) [reply www/css/dark.css:8]: They are grayed out as an inidcation that their configuration is empty.

This property is currently handled via a direct `style` attribute, so it should be changed to a CSS class.

File: www/css/dark.css (hunk 8 above)
Comment (Alexejhero) [reply www/css/dark.css:29]: I know that, but the contrast between them and the dark background is different than the contrast between them and the light background. 

In my opinion they look more pale on the dark background (I may just be seeing things.)

File: www/css/dark.css (hunk 6 above)
Comment (WilsontheWolf) [reply www/css/dark.css:19]: Shadows? I can't find any shadows.
light
dark

File: www/css/dark.css (hunk 7 above)
Comment (WilsontheWolf) [reply www/css/dark.css:24]: Bars

File: www/css/dark.css (hunk 5 above)
Comment (WilsontheWolf) [reply www/css/dark.css:8]: image

File: src/init.js [hunk 10]
```diff
@@ -14,11 +14,12 @@ module.exports = function () {
 
 function initialize() {
   setInterval(() => {
-    http.get(`http://dlockly.glitch.me/`);
+    http.get(`http://${process.env.PROJECT_DOMAIN}.glitch.me/`);
```
Comment (WilsontheWolf) [inline src/init.js:17]: Made it easier for me (or anyone) to add it. now it uses a custom glitch domain so I can easily run it and dont need to turn it back

File: src/requests/login.js [hunk 11]
```diff
@@ -5,5 +5,5 @@ const server = require('../../server');
 
 module.exports = function (data) {
   if (auth.sessionValid(data.authUserID, data.authSession)) data.res.redirect("/");
-  else data.res.redirect("https://discordapp.com/api/oauth2/authorize?client_id=591694201230721043&redirect_uri=https%3A%2F%2Fdlockly.glitch.me%2Fauth&response_type=code&scope=identify");
+  else data.res.redirect("https://discordapp.com/api/oauth2/authorize?client_id="+server.bot.user.id+"&redirect_uri=https%3A%2F%2F"+process.env.PROJECT_DOMAIN+".glitch.me%2Fauth&response_type=code&scope=identify");
```
Comment (WilsontheWolf) [inline src/requests/login.js]: Also the same as other comment but made the id work too. This means that the login page won't work if the bot is not signed in

File: www/css/main.css [hunk 12]
```diff
@@ -73,7 +73,6 @@ svg#login {
   bottom: 0;
   width: 100%;
   height: 38px;
-  background-color: white;
```
Comment (WilsontheWolf) [inline www/css/main.css]: Fixes footer wrong color

File: www/html/browserunsup.ejs [hunk 13]
```diff
@@ -1,3 +1,4 @@
 <% include partials/head %>
 
-<h1>Please use <a href='https://www.google.com/chrome/'>Google Chrome</a> or <a href='https://www.mozilla.org/en-US/firefox/new/'>Mozilla Firefox</a> in order to view this webpage</h1>
\ No newline at end of file
+<h1>Please use <a href='https://www.google.com/chrome/'>Google Chrome</a> or <a href='https://www.mozilla.org/en-US/firefox/new/'>Mozilla Firefox</a> in order to view this webpage</h1>
+<% include partials/footer %>
```
Comment (WilsontheWolf) [inline www/html/browserunsup.ejs:5]: Added footer

File: www/html/maintenance.ejs [hunk 14]
```diff
@@ -1,3 +1,5 @@
 <% include partials/head %>
 
 <h1>Under maintenance</h1>
+
+<% include partials/footer %>
```
Comment (WilsontheWolf) [inline www/html/maintenance.ejs:5]: Added footer

File: www/js/cookies.js [hunk 15]
```diff
@@ -1,9 +1,8 @@
 'use strict';
 
 function setCookie(cname, cvalue, exdays = 365) {
-  var d = new Date();
-  d.setTime(d.getTime() + (exdays * 24 * 60 * 60 * 1000));
-  var expires = "expires=" + d.toUTCString();
+  
+  var expires = "expires=" + new Date(Date.now() +exdays * 24 * 60 * 60 * 1000).toUTCString();
```
Comment (WilsontheWolf) [inline www/js/cookies.js]: made this take less room

**Comment by Alexejhero** [comment]
~~Why did you override 99% of my commits?~~

Nevermind, git was showing me things that weren't real.

File: www/html/browserunsup.ejs (hunk 13 above)
Comment (Alexejhero) [reply www/html/browserunsup.ejs:5]: I intentionally did not add the footer here.

Comment (Alexejhero) [reply www/html/browserunsup.ejs:5]: Actually nevermind.

File: www/css/dark.css (hunk 6 above)
Comment (Alexejhero) [reply www/css/dark.css:19]: Shadows are blocks that don't actually exist. An example is the string block in the send message block.

Comment (Alexejhero) [reply www/css/dark.css:19]: Shadow selector is `.blocklyPath[fill='#bddbd1']`

File: src/init.js [hunk 16]
```diff
@@ -18,10 +18,10 @@ function initialize() {
   }, 280000);
 
   server.bot.on("ready", () => {
-    server.bot.user.setActivity("with blocks. https://"+process.env.PROJECT_DOMAIN+".glitch.me");
```
Comment (WilsontheWolf) [inline src/init.js]: this was unnessicary.
